### PR TITLE
Expand the city dialog text for wonders

### LINKED ISSUE #269: The city dialog should display how many citizens can be made content with existing buildings
Currently, the citizens made content by buildings are shown, but one can't see how much the city can grow before the first unhappy citizen appears. This makes celebration harder.

Initial proposal is to add it to the relevant tooltip in the Happiness tab. In the long run some more graphical depiction could be added.

### PR DESCRIPTION
**Pull request by lmoureaux**
Improve the description by explaining which effects are active and, for two of
them, what contributes to their value (the third one is a simple binary effect,
and that's harder to display properly). Helps #269.

Factor out a function to build such effect descriptions so we don't need to
repeat the same code over and over again.

Granted, I wasn't very inspired when writing the default text.

With this PR, I'm happy to close #269, so:
Closes #269.

### CODE REVIEW

**Comment by daavko** [comment]
Far too manu occurences to list: "citizen in this city" instead of "citizen of this city".

**Comment by lmoureaux** [comment]
> Far too many occurences to list: "citizen in this city" instead of "citizen of this city".

Done

File: client/text.cpp [hunk 1]
```diff
@@ -1459,23 +1474,63 @@ QString text_happiness_nationality(const struct city *pcity)
...
+  bool wrote_something = false;
 
-    effects = get_effect_list_req_text(plist);
-    str = QString(_("Wonders: %1.")).arg(effects);
-  } else {
-    str = _("Wonders: None.");
+  int happy_bonus = get_city_bonus(pcity, EFT_MAKE_HAPPY);
+  if (happy_bonus > 0) {
+    if (happy_bonus == 1) {
+      str += _("Up to one content citizen in this city is made happy.");
```
Comment (daavko) [inline client/text.cpp:1491]: Consider "One content citizen in this city can be made happy." instead

File: client/text.cpp [hunk 2]
```diff
@@ -1459,23 +1474,63 @@ QString text_happiness_nationality(const struct city *pcity)
...
+    } else {
+      str +=
+          QString(PL_("Up to %1 unhappy or content citizen in this city is "
+                      "made happy (unhappy citizens count double).",
+                      "Up to %1 unhappy or content citizens in this city "
+                      "are made happy (unhappy citizens count double).",
+                      happy_bonus))
```
Comment (daavko) [inline client/text.cpp:1498]: Consider "can be made happy" instead of "is/are made happy"

File: client/text.cpp [hunk 3]
```diff
@@ -1459,23 +1474,63 @@ QString text_happiness_nationality(const struct city *pcity)
...
+    str += text_happiness_effect_details(pcity, EFT_MAKE_HAPPY);
+    str += QStringLiteral("</p><p>");
+    wrote_something = true;
   }
-  effect_list_destroy(plist);
 
-  return str.trimmed();
+  if (get_city_bonus(pcity, EFT_NO_UNHAPPY) > 0) {
+    str += _("No citizens in this city can ever be unhappy or angry. "
+             "Any unhappy or angry citizen is automatically made content.");
```
Comment (daavko) [inline client/text.cpp:1511]: Consider "... or angry citizen*s are* automatically ..." instead, to be consistent with the first sentence.

File: client/text.cpp [hunk 4]
```diff
@@ -1459,23 +1474,63 @@ QString text_happiness_nationality(const struct city *pcity)
...
+  } else if (int bonus = get_city_bonus(pcity, EFT_FORCE_CONTENT);
+             bonus > 0) {
+    str += QStringLiteral("</p><p>");
+    str += QString(PL_("Up to %1 unhappy or angry citizen in this city is "
+                       "made content.",
+                       "Up to %1 unhappy or angry citizens in this city are "
+                       "made content.",
```
Comment (daavko) [inline client/text.cpp:1518]: Consider "can be made content" instead of "is/are made content"

**Review by jwrober — APPROVED** [review]
Looks good to me.

**Pull request merged**